modules/nf-core/mitohifi/mitohifi/meta.yml

```diff
@@ -25,6 +25,10 @@ input:
       type: file
       description: Reference annotation
       pattern: "*.{gb}"
+  - input_mode:
```

Comment:
I would change this to `use_reads` and have it be a boolean value. This means that `input_mode` doesn't need to be a string that has to be checked for it's value, and if it's true, use `-r`, otherwise `-c`. It would remove the need for the error message too.

Reply:
I'd rather keep it as is to opt for readability in the module calls. Also, boolean version suggests there should be a default value, while in this case setting the value explicitly is preferable. Also as @muffato points out boolean value is more off and on in its nature, while this case we're choosing between two different workflow types.

Reply:
Sorry, I'm not sure I understand. Can you show me how you're using it? 
In a module call, you can also use variables or comments to enhance readability. 
```nextflow
use_reads = true // could potentially be a params too. 
SOME_PROCESS (
    ch_input,
    use_reads // true: input are reads, false: input are contigs
)
```
In the current state, one needs to add a check in the module that `input_mode` should be either `-r` or `-c`.

I think this case is not different from https://github.com/nf-core/modules/blob/8fc1d24c710ebe1d5de0f2447ec9439fd3d9d66a/modules/nf-core/bwa/mem/main.nf#L13 for example.

Reply:
I do see a difference with flags such as this `sort_bam`. I read booleans as extra steps that are typically off by default and explicitly turned on. In the case of MitoHifi, it's not an on/off thing, it's a alternative mode. I think a more accurate comparison is the NCBI datasets module (#4047) which does different things depending on the mode

Reply:
OK. I get the idea better. What about then making the modes just `r` or `c` and having the code as `-$mode`.